blocks/library/gallery/block.js

```diff
@@ -51,33 +52,38 @@ class GalleryBlock extends Component {
 
 		this.state = {
```

Comment:
Do you think we could move this entire state to `GalleryImage`? It doesn't really seem needed this far up, and we'd avoid passing so much down the component tree.

Reply:
I think it would also avoid `onSelectCaption` being a wrapped function.

Reply:
Hi @iseulde, I agree that it makes sense to move the captionSelected state to the GalleryImage and I updated the code. Regarding the index of the currently selected image I kept it in Gallery block because when we select an image we automatically unselect another image,  that index affects the state of multiple images so it makes sense to stay in the parent.